addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3215,7 +3294,31 @@ class SnippetsMenu extends Component {
                     snippet.isRenaming = false;
                 }
 
-                this.snippets.set(snippet.id, snippet);
+                this.snippets.set(snippet.key, snippet);
+            }
+            const snippets = Array.from(this.snippets.values());
+            const customSnippets = snippets.filter(snippet => {
+                return snippet.category.id === "snippet_custom";
+            });
+            for (const customSnippet of customSnippets) {
+                // The "s_button" has a numeric value added to its name when it
+                // is custom, so we need to consider this in the search.
+                const customSnippetName = /s_button_\d+/.test(customSnippet.name) ?
+                    "s_button" :
+                    customSnippet.name;
+
+                const categoryIds = Object.values(snippets)
```

Comment:
`snippets` is already an array, no need to go through `Object.values()` (which doesn't do anything here I think?)